tastytrade/market_sessions.py

```diff
@@ -0,0 +1,145 @@
+from datetime import date, datetime
+from typing import Optional, List
+
+from tastytrade.session import Session
+from tastytrade.utils import TastytradeJsonDataclass
+
+
+class MarketTimeSessionsNext(TastytradeJsonDataclass):
+    """Dataclass for data inside the 'next-session' field."""
+    close_at: datetime
+    close_at_ext: Optional[datetime] = None
+    instrument_collection: str
+    open_at: datetime
+    session_date: date
+    start_at: datetime
+
+
+class MarketTimeSessionsPrevious(TastytradeJsonDataclass):
+    """Dataclass for data inside the 'previous-session' field."""
+    close_at: datetime
+    close_at_ext: Optional[datetime] = None
+    instrument_collection: str
+    open_at: datetime
+    session_date: date
+    start_at: datetime
+
+
+class MarketTimeSessionsCurrent(TastytradeJsonDataclass):
+    """
+    Dataclass representing the current session and any nested
+    'next' or 'previous' session info.
+
+    NOTE: The JSON you showed places `close-at` etc. under
+          next-session/previous-session, not at the top level.
+          Here, we treat 'close_at', 'open_at', etc. as the
+          ones from 'next_session', purely as an example.
+    """
+    close_at: Optional[datetime] = None
+    close_at_ext: Optional[datetime] = None
+    instrument_collection: str
+    open_at: Optional[datetime] = None
+    start_at: Optional[datetime] = None
+    next_session: Optional[MarketTimeSessionsNext] = None
+    previous_session: Optional[MarketTimeSessionsPrevious] = None
+    state: str = ""
+
+
+class MarketTimeSessionsResponse(TastytradeJsonDataclass):
+    """
+    Top-level container for the API response, which holds
+    multiple MarketTimeSessionsCurrent items.
+    """
+    items: List[MarketTimeSessionsCurrent]
+
+
+class MarketCalendarData(TastytradeJsonDataclass):
+    """
+    Represents a data structure holding market half-days and holidays,
+    each being a list of ISO-format date strings that get parsed into Python 'date' objects.
+    """
+    market_half_days: List[date]
+    market_holidays: List[date]
+
+
+async def a_get_market_time_sessions(session: Session, instrument_collections: list[str]) -> list[MarketTimeSessionsCurrent]:
```

Comment:
Can you rename instrument_collections to "exchanges" and create an string enum with the valid options? Also let's rename "Equities" in the enum to NYSE